3: Add linters configuration

EugeneTurkin opened the pull request:
Parent story: https://github.com/fenya123/bingin/issues/9

Here at **fenya123** we don't settle for anything other than perfection. That is why our ACIH's backend code has to be uniform and consistent from start to finish. To achieve that, we are going to use linters: `mypy`, `ruff`, `flake8`, `pylint`.

In the world of coding speed is _everything_. The slower you are, the faster you and your work become _obsolete_. Following linters have overlapping checks and also vary in speed (_fastest_ to **slowest**): `ruff`, `flake8`, `pylint`.

Naturally, _the need for speed_ necessitates us to get rid of overlapping checks. We'll achieve that by configuring faster linters to perform as many checks as possible, while slower linters will only perform checks that others linters can't.

In the scope of this task we'll need to:
1) add and configure our linters of choice and their plugins,
2) add instructions to `README.md` on how to run them.

birthdaysgift commented on pyproject.toml:
```diff
@@ -13,3 +13,10 @@ version = "0.1.0"
 python = "3.11.*"
+
+
+[tool.poetry.group.dev.dependencies]
```
This group should be optional, because we don't want these dependencies to be installed by default with:
```python
poetry install
```

We want to install this group only where we actually need it, like this
```python
poetry install --with dev
```

birthdaysgift commented on .pylintrc:
Pylint gives me an error when I try to run it. We need to figure it out and fix

EugeneTurkin committed:
Add linters configuration (#2)

Parent story: https://github.com/fenya123/bingin/issues/9

Here at fenya123 we don't settle for anything other than perfection. That is why our ACIH's backend code has to be uniform and consistent from start to finish. To achieve that, we are going to use linters: mypy, ruff, flake8, pylint.

In the world of coding speed is everything. The slower you are, the faster you and your work become obsolete. Following linters have overlapping checks and also vary in speed (fastest to slowest): ruff, flake8, pylint.

Naturally, the need for speed necessitates us to get rid of overlapping checks. We'll achieve that by configuring faster linters to perform as many checks as possible, while slower linters will only perform checks that others linters can't.

In the scope of this task we'll need to:

add and configure our linters of choice and their plugins,
add instructions to README.md on how to run them.
